pkg/config/types/default_config.go

```diff
@@ -134,13 +137,33 @@ const defaultBacalhauDir = ".bacalhau"
 // 2. User's home directory with .bacalhau appended.
 // 3. If all above fail, use .bacalhau in the current directory.
 func DefaultDataDir() string {
+	// this method runs before root.go, so we set the level to info for these calls, then return it to previous value
+	currentLevel := zerolog.GlobalLevel()
+	zerolog.SetGlobalLevel(zerolog.InfoLevel)
+	defer zerolog.SetGlobalLevel(currentLevel)
+
+	// Check if the BACALHAU_DIR environment variable is set
```

Comment:
my concern with the warning logs you introduced here is there is no way for the user to avoid them, except by setting the env variable. If they run `--data-dir <path>` they will still get warnings if they don't have home directory which doesn't make sense.

This method is about figuring out the default path, and it shouldn't fail if it couldn't. The config validation should fail later

Reply:
We only warn if the user set the env var to an empty value, a rare case, but possible. Would you prefer we remove the log line?

Reply:
Not true. You warn when you can't find the home directory and when you can't figure out a default repo

Reply:
My previous comment was in the context of the line you commented on. True, for other cases we log a warning: If we cannot determine a users home directory, and thus the default location where bacalhau initializes/opens it repo this seems worth a warning. But it sounds like you rather we remove all these log lines, is that correct?